packages/frontend/src/components/Election/Admin/AdminHome.tsx

```diff
@@ -470,6 +471,7 @@ const AdminHome = () => {
             })
         if (!confirmed) return
         try {
+            await deleteAllBallots()
```

Comment:
Deleting ballots during finalization should be done in the back end as a part of the finalize election controller otherwise this would be easy to bypass.

Reply:
Bypassing shouldn't be a problem since the backend has a check to only allow the operation if the election is in draft. 

Moving deleteAllBallots to the backend for finalize would be pretty easy, but the deleteAllBallot calls for updating races would be a little more awkward. Since updating a race applies an update to the whole election, the backend would have to check which part of the structure was updated in order to determine if the ballots need to be deleted.

Reply:
My concern is that someone could finalize the election without clearing the ballots.

Reply:
For dealing with updating the race, maybe we could have a parameter in the update election request that indicates what's being changed.

Reply:
Updated the PR

During finalize the deleteAllBallots is called from the backend, but for adjusting races I left that in the frontend.

I figure as long as they're guaranteed to reset during finalize, it's not a problem if someone bypasses it for adding races. I'm sure we'll come up with a more general solution later though.